input/elasticapm/internal/modeldecoder/v2/decoder.go

```diff
@@ -361,26 +360,28 @@ func mapToCloudModel(from contextCloud, cloud *modelpb.Cloud) {
 
 func mapToClientModel(from contextRequest, source **modelpb.Source, client **modelpb.Client) {
 	// http.Request.Headers and http.Request.Socket are only set for backend events.
-	if _, err := netip.ParseAddr((*source).GetIp()); err != nil {
+	if (*source).GetIp() == nil {
 		ip, port := netutil.SplitAddrPort(from.Socket.RemoteAddress.Val)
```

Comment:
We could remove one parsing here if we stop relying on netutil, and split the IP/Port from a string directly into an IP.